### PR TITLE
Markdown Documentation with mkdocs

### PR DESCRIPTION
**Pull request by Leo-MathGuy**
Opening as draft until finished, for feedback and stuff like that

There are a lot of things outside the README in the pyndustric, so I think documentation is needed for.

What will do:

- API Documentation (copying pyndustri.pyi into readable format)
- Examples

I already finished setup and supported features (mostly copied from the README but changed a few things) and the reference for Env.

My fork is autodeploying to [netlify](https://pyndustric.netlify.app/)

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+name: Build and Deploy Docs to Netlify
```
Comment (Lonami) [inline .github/workflows/main.yml:1]: If we go this route, I would probably just push the documentation to gh-pages or perhaps readthedocs.

File: doc/docs/api/env.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
+# Environmental variables
```
Comment (Lonami) [inline doc/docs/api/env.md:1]: Maintaining this separate from the code is very error prone. We very likely want to generate it from the code itself.

File: doc/docs/setup.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
+# Setup
+
+## Installation
+
+### Install via pip:
+
+`pip3 install https://github.com/Lonami/pyndustric/archive/refs/heads/master.zip`
```
Comment (Lonami) [inline doc/docs/setup.md:7]: Inline code blocks are likely not the right choice, rather, I'd recommend:

````md
```sh
pip install ....
```
````

File: doc/docs/setup.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+or
+
+`python3 -m pip install https://github.com/Lonami/pyndustric/archive/refs/heads/master.zip`
+
+### Install from source
+
+To install from the source code (for development or something)
```
Comment (Lonami) [inline doc/docs/setup.md:15]: I can see where this is coming from but I'm also not sure we should explain "how to use Python". Maybe it's better to link to setuptools documentation.

File: doc/docs/supported.md [hunk 5]
```diff
@@ -0,0 +1,174 @@
+# Supported Features
```
Comment (Lonami) [inline doc/docs/supported.md:1]: Repetition with the README is unfortunate.

File: .github/workflows/main.yml (hunk 1 above)
Comment (Leo-MathGuy) [reply .github/workflows/main.yml:1]: I will setup pushthedocs a bit later then

File: doc/docs/setup.md (hunk 4 above)
Comment (Leo-MathGuy) [reply doc/docs/setup.md:15]: You're right, if someone does need to do that they are probably a developer already know how to setup a module

**Comment by Leo-MathGuy** [comment]
Ok, I used pydoc-markdown with mkdocs to generate the docs. I'll add some examples

**Comment by Leo-MathGuy** [comment]
Wait no, It generated the wrong thing, not the pyi file

**Pull request closed without merging**